src/components/Form/index.tsx

```diff
@@ -0,0 +1,86 @@
+import RjsForm from '@rjsf/mui';
+import ajvValidator from '@rjsf/validator-ajv8';
+import { FormProps as RjsFormProps } from '@rjsf/core';
+import { useCallback } from 'react';
+import { Box, Button, ButtonProps } from '@mui/material';
+import { PasswordWidget } from './PasswordWidget';
+import { TextWidget } from './TextWidget';
+import { TextareaWidget } from './TextareaWidget';
+import { SelectWidget } from './SelectWidget';
+import { DateTimeWidget } from './DateTimeWidget';
+import { UpDownWidget } from './UpDownWidget';
+
+const internalWidgets: {
+	[k: string]: any;
+} = {
+	PasswordWidget,
+	EmailWidget: TextWidget,
+	TextWidget,
+	TextareaWidget,
+	SelectWidget,
+	UpDownWidget,
+	DateTimeWidget,
+};
+
+export interface MuiFormProps extends Omit<RjsFormProps, 'validator'> {
+	/** If true, do not display the form submit button */
+	hideSubmitButton?: boolean;
+	/** Properties that are passed to the submit button, these are the same props used for the [`Button`](#button) component */
+	submitButtonProps?: ButtonProps;
+	/** If passed, it will show a secondary button, these are the same props used for the [`Button`](#button) component */
+	secondaryButtonProps?: ButtonProps;
+
+	validator?: RjsFormProps['validator'];
+}
+
+export const MuiForm: React.FC<React.PropsWithChildren<MuiFormProps>> = ({
+	hideSubmitButton,
+	submitButtonProps,
+	secondaryButtonProps,
+	validator = ajvValidator,
+	widgets,
+	fields,
+	children,
+	...otherProps
+}) => {
+	const handleKeyUpDown = useCallback(
+		(e: React.KeyboardEvent<HTMLDivElement>) => {
+			if (e.key === 'Enter' && !hideSubmitButton) {
+				e.stopPropagation();
+			}
+		},
+		[hideSubmitButton],
+	);
```

Comment:
As noted in the comment about `hideSubmitButton`, we should discuss the appropriate way to enable submit on enter with native behavior instead of these workarounds we're making (I know this comes from rendition, just want to improve and do things right moving forward)

Reply:
This is not needed for the Form itself, form submit button would work with this without problems. I pretty sure this was added at some point as we have the dashboard showing a modal with a components that override this behaviour. We can try to remove it and try to fix the dashboard where necessary

Reply:
> This is not needed for the Form itself, form submit button would work with this without problems. 

Doesn't work when `hideSubmitButton` is true. So this useEffect is used for that case as well. As noted earlier, this shouldn't be a case either ...

Reply:
> We can try to remove it and try to fix the dashboard where necessary

We should 💯